### ISSUE TITLE
Avatar Animations

### ISSUE THREAD

**Issue by rchadwic** [adlnet/Sandbox#276]
Can you direct us to some documentation about the best workflow for creating avatars for the sandbox? From what I can see, right now the standard avatars are just collada with a simple walk animation on them.Is key frame animation system with three.js is the way to go? Is this already built into the sandbox? Right now, we can only do one walk animation. Is there a way to support more animations and even sit down?

**Comment by rchadwic** [comment]
Yes - the Avatar is really just a model. If you were to modify the source of the model, you can have more frames in there. Select your avatar and open the script editor. Look for the 'cycles' property. (You might have to enable "show inherited properties". This is the list of cycles - it's just a range from the full keyframed animation. 

The code that calls different animation cycles at different times is public/adl/sandbox/character.vwf.yaml. This would require some additional work to make a different cycle play on a keypress or something. This bit of code is absolutely ancient and I'd love for someone to update it and make it more extensible.

**Comment by wilo1** [comment]
Yes I see where that is controlled now. The question now is how edit it to recognize different animations in the sequence. The current sandbox avatar (well most of them) can sit down yet the control keys do not have that option the character.vwf.yaml. Technically, if we create an avatar that has various animations it should now just be a matter of setting up the controller correctly. That is where I am stuck. I will keep researching ways to do this but if there are any resources you can point me to it would be appreciated.
![image](https://cloud.githubusercontent.com/assets/6297072/10634357/2e630140-77bf-11e5-867e-65bb4ecf1115.png)

**Comment by rchadwic** [comment]
Yeah, we really need a way to like set a custom cycle, and play that. Maybe we can expose a custom cycle in the cycle list that can have it's start and end times set by script. Then, that cycle could override the defaults ones. That actually sounds reasonably easy to implement.... let me think on it...

**Comment by wilo1** [comment]
Is it possible to edit the character.vwf.yaml file with new cycle actions or is that done in a different place? Right now it has the basics like walk jump etc. If the character is created with specific named animations and they are added to the system would that work? For instance edit the existing cycles to reflect new ones we add?

 this.runningjump = function() {

        if (this.inMotion('jump')) return;
        this.addMotion('jump');
        this.jumpcounter = 0;
        this.resetCycle(this.cycles.runningjump);
        this.removeCycle('runningjump');
        this.addCycle('runningjump');

**Comment by rchadwic** [comment]
Yeah. If you add the cycle names and edit all those values, you should be able to add another cycle. But, this really needs to be refactored to be more general. Even if you add a cycle to the cycles property, you'll still need some way to add that to a key press. These key presses are sort of hard coded to cycles, and that's where this needs to be updated.
